react-bootstrap-table/react-bootstrap-table.d.ts

```diff
@@ -520,11 +558,13 @@ declare module "react-bootstrap-table" {
 		options?: any;
 	}
 	
+	export type FilterType = 'TextFilter' | 'RegexFilter' | 'SelectFilter' | 'NumberFilter' | 'DateFilter' | 'CustomFilter';
```

Comment:
OTOH, the nice thing about declaring `FilterType` is that a good editor/IDE gives you code completion. If someone wants to use a custom filter, they could use

```typescript
{
     filter: 'MyCustomFilter' as FilterType
}
```

Which would keep the type checking and the code completion while making it explicit that you are using a custom filter...